pkg/util/utils.go

```diff
@@ -244,3 +250,104 @@ func UpdateK8sResourceAnnotation(client client.Client, ctx *context.Context, k8s
 	}
 	return nil
 }
+
+// some resources has complex index, so set it type to string
+func GenerateID(res_id, prefix, suffix string, index string) string {
+	var id strings.Builder
+	if len(prefix) > 0 {
+		id.WriteString(prefix)
+		id.WriteString("_")
+	}
+
+	id.WriteString(res_id)
+	if len(index) > 0 {
+		id.WriteString("_")
+		id.WriteString(index)
+
+	}
+	if len(suffix) > 0 {
+		id.WriteString("_")
+		id.WriteString(suffix)
+	}
+	return id.String()
```

Comment:
we have NormalizeId func, 
https://github.com/vmware-tanzu/nsx-operator/blob/vpc_dev/pkg/util/utils.go#L57
do we need to generate ID suing this func?

Reply:
NormalizeId only used for truncate ID.